sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -0,0 +1,444 @@
+package com.salesforce.rules;
+
+import com.salesforce.apex.jorje.ASTConstants;
+import com.salesforce.apex.jorje.ASTConstants.NodeType;
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.exception.UnexpectedException;
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.build.CaseSafePropertyUtil.H;
+import com.salesforce.graph.ops.PathEntryPointUtil;
+import com.salesforce.graph.ops.directive.EngineDirective;
+import com.salesforce.graph.vertex.InvocableWithParametersVertex;
+import com.salesforce.graph.vertex.MethodVertex;
+import com.salesforce.graph.vertex.SFVertexFactory;
+import com.salesforce.graph.vertex.UserClassVertex;
+import com.salesforce.rules.unusedmethod.CallValidator;
+import com.salesforce.rules.unusedmethod.InternalCallValidator;
+import com.salesforce.rules.unusedmethod.SubclassCallValidator;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+public class UnusedMethodRule extends AbstractStaticRule {
```

Comment:
Doesn't have to be in this iteration - can we refactor this class a little to break it into more focused parts? Doesn't have to be anything fancy - just smaller groups of similar methods. This class is about to grow even larger and already feels a little overwhelming.